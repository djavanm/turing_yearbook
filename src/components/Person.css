img {
  height: 250px;
  width: 250px;
  border-radius: 5px;
  margin: 5px;
}

article {
  background: #ffffff;
  border-radius: 5px;
  box-sizing: border-box;
  padding: 10px;
  margin: 10px;
  box-shadow: -7px 5px 10px -2px rgba(0,0,0,0.59);
  font-size: 1.3em;
}

