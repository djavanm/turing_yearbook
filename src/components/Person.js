import React, { Component } from "react";
import "./Person.css";

export default class Person extends Component {
  constructor(props) {
    super(props);
    this.state = {
      id: props.person.id,
      name: props.person.name,
      quote: props.person.quote,
      superlative: props.person.superlative,
      photo: props.person.photo
    };
  }
  updateInput = (e, prop) => {
    this.setState({
      [prop]: e.target.innerText
    });
    this.props.updatePerson(this.state);
  };

  clearInput = e => {
    if (this.props.editable) {
      e.target.innerText = "";
    }
  };

  render() {
    const { id, photo, name, quote, superlative } = this.state;
    const { editable, removePerson } = this.props;
    return (
      <article data-id={id}>
        <img src={photo} alt={`${name}'s avatar`} />
        <h3
          contentEditable={editable}
          onClick={e => this.clearInput(e)}
          onBlur={e => this.updateInput(e, "name")}
          suppressContentEditableWarning={true}
        >
          {name === "" ? "Enter a Name" : `{ ${name} }`}
        </h3>
        <p
          contentEditable={editable}
          onClick={e => this.clearInput(e)}
          onBlur={e => this.updateInput(e, "quote")}
          suppressContentEditableWarning={true}
        >
          {quote === "" ? "Enter a Quote" : quote}
        </p>
        <h4
          contentEditable={editable}
          onClick={e => this.clearInput(e)}
          onBlur={e => this.updateInput(e, "superlative")}
          suppressContentEditableWarning={true}
        >
          {superlative === "" ? "Enter a Superlative" : superlative}
        </h4>
        {this.props.editable && (
          <button className="button" onClick={() => removePerson(this.state)}>
            <span className="visible">{`{ `}</span>
            Delete
            <span className="visible">{`  }`}</span>
          </button>
        )}
      </article>
    );
  }
}
