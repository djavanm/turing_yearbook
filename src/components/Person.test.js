import React from 'react';
import { shallow } from 'enzyme';
import Person from './Person';

let testPerson = {id: 1, name: 'Eric', quote: 'You miss all the shots you don\'t take.', superlative: 'Most Likely to Win the PGA', photo: './eric.jpg'}
let testStudent = {id: 21, name: 'Abigail', quote: '', superlative: '', photo: 'https://placekitten.com/200/300'}

describe('Person', () => {
  it('matches the staff snapshot created for Person ', () => {
    const updatePersonMock = jest.fn();
    const removePersonMock = jest.fn();
    const wrapper = shallow(<Person person={testPerson} 
        key={testPerson.id} 
        editable={false} 
        updatePerson={updatePersonMock} 
        removePerson={removePersonMock}/>);

    expect(wrapper).toMatchSnapshot();
  });
  it('matches the student snapshot created for Person ', () => {
    const updatePersonMock = jest.fn();
    const removePersonMock = jest.fn();
    const wrapper = shallow(<Person person={testStudent} 
        key={testStudent.id} 
        editable={true} 
        updatePerson={updatePersonMock} 
        removePerson={removePersonMock}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('calls the remove person method on App, on click', () => {
    const updatePersonMock = jest.fn();
    const removePersonMock = jest.fn();
    const wrapper = shallow(<Person person={testStudent} 
        key={testStudent.id} 
        editable={true} 
        updatePerson={updatePersonMock} 
        removePerson={removePersonMock}/>);

    wrapper.find('button').simulate('click');
    expect(removePersonMock).toHaveBeenCalledWith(testStudent)
  });
});