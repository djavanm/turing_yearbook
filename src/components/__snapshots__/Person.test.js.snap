// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person matches the staff snapshot created for Person  1`] = `
<article
  data-id={1}
>
  <img
    alt="Eric's avatar"
    src="./eric.jpg"
  />
  <h3
    contentEditable={false}
    onBlur={[Function]}
    onClick={[Function]}
    suppressContentEditableWarning={true}
  >
    { Eric }
  </h3>
  <p
    contentEditable={false}
    onBlur={[Function]}
    onClick={[Function]}
    suppressContentEditableWarning={true}
  >
    You miss all the shots you don't take.
  </p>
  <h4
    contentEditable={false}
    onBlur={[Function]}
    onClick={[Function]}
    suppressContentEditableWarning={true}
  >
    Most Likely to Win the PGA
  </h4>
</article>
`;

exports[`Person matches the student snapshot created for Person  1`] = `
<article
  data-id={21}
>
  <img
    alt="Abigail's avatar"
    src="https://placekitten.com/200/300"
  />
  <h3
    contentEditable={true}
    onBlur={[Function]}
    onClick={[Function]}
    suppressContentEditableWarning={true}
  >
    { Abigail }
  </h3>
  <p
    contentEditable={true}
    onBlur={[Function]}
    onClick={[Function]}
    suppressContentEditableWarning={true}
  >
    Enter a Quote
  </p>
  <h4
    contentEditable={true}
    onBlur={[Function]}
    onClick={[Function]}
    suppressContentEditableWarning={true}
  >
    Enter a Superlative
  </h4>
  <button
    className="button"
    onClick={[Function]}
  >
    Delete
  </button>
</article>
`;
